feat(docs): Create JobSet adopters page (#852)
kubernetes-sigs/jobset: ADOPTERS.md

> Below is a list of organizations that use JobSet in production along with the integrations they use.

Reviewer: ```suggestion
Below is a list of organizations and projects that use JobSet in production along with the integrations they use.
```

If you are using JobSet, feel free to open a pull request to add your organization to the list.

## Adopters

| Organization                                            | Description                               | Contact                                          |
| ------------------------------------------------------- | ----------------------------------------- | ------------------------------------------------ |

> | [Kubeflow Trainer](https://github.com/kubeflow/trainer) | Core component for TrainJob orchestration | [@andreyvelich](https://github.com/andreyvelich) |

Reviewer 2: ```suggestion
| [Kubeflow Trainer](https://github.com/kubeflow/trainer) | Low-level runtime for distributed ML training on Kubernetes | [@andreyvelich](https://github.com/andreyvelich) |
```

Shall we update the description to provide users with more straightforward insights? They may not understand what is the TrainJob.

Reviewer 3: +1 on this.

Reviewer: +1, could also be "Job management for distributed AI/ML training". I guess Kubernetes can be omitted.

Author: Job management for distributed AI/ML training sounds good.

> | [Kubeflow Trainer](https://github.com/kubeflow/trainer) | Job management for distributed AI/ML training | [@andreyvelich](https://github.com/andreyvelich) |

Reviewer 4: we can add google
```suggestion
| [Kubeflow Trainer](https://github.com/kubeflow/trainer) | Job management for distributed AI/ML training | [@andreyvelich](https://github.com/andreyvelich) |

| [Google TPU Training] (https://cloud.google.com/kubernetes-engine/docs/tutorials/tpu-multislice-kueue) | Large scale distributed AI/ML training on TPUs | [@ahg-g](https://github.com/ahg-g) |

```
